sandbox/modules/identity/src/main/java/org/opensearch/identity/configuration/DynamicConfigFactory.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity.configuration;
+
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.opensearch.authn.User;
+import org.opensearch.identity.configuration.model.InternalUsersModel;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.LogManager;
+import org.opensearch.client.Client;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.threadpool.ThreadPool;
+import org.greenrobot.eventbus.EventBus;
+import org.greenrobot.eventbus.EventBusBuilder;
+
+public class DynamicConfigFactory implements Initializable, ConfigurationChangeListener {
+
+    public static final EventBusBuilder EVENT_BUS_BUILDER = EventBus.builder();
```

Comment:
I like the usage of event bus for this rather than rolling our own, but that said have you seen how this is done elsewhere in OpenSearch?

Reply:
Do you know of examples within core of doing async programming to listen for changes on an index? Possibly `IndexOperationListener`?

One of the challenges is making sure that nodes are synchronized and the method introduced here reloads the index into memory every 3s using the event bus.

Reply:
@saratvemulapalli It looks like you fielding ad-hoc requests this week - did you have any recommendations for this sub/pub processes in OpenSearch?